2022-07-28 11:21:48,190 INFO     ---------------------------------------------------------------------------------------------
2022-07-28 11:21:48,190 INFO     Geo: cqd
2022-07-28 11:21:48,190 INFO     Data Path: data/FB15k-237-betae
2022-07-28 11:21:48,190 INFO     #entity: 14505
2022-07-28 11:21:48,190 INFO     #relation: 474
2022-07-28 11:21:48,190 INFO     #max steps: 150000
2022-07-28 11:21:48,190 INFO     Evaluate unoins using: DNF
2022-07-28 11:21:48,190 INFO     loading data
2022-07-28 11:22:30,088 INFO     Training info:
2022-07-28 11:22:30,092 INFO     2p: 149689
2022-07-28 11:22:32,510 INFO     Validation info:
2022-07-28 11:22:32,510 INFO     2p: 5000
2022-07-28 11:22:32,512 INFO     Test info:
2022-07-28 11:22:32,512 INFO     2p: 5000
2022-07-28 11:22:37,995 INFO     Model Parameter Configuration:
2022-07-28 11:22:37,995 INFO     Parameter embeddings.0.weight: torch.Size([14505, 2000]), require_grad = False
2022-07-28 11:22:37,995 INFO     Parameter embeddings.1.weight: torch.Size([474, 2000]), require_grad = False
2022-07-28 11:22:37,995 INFO     Parameter score_transform_ent_fun.weight: torch.Size([2, 2000]), require_grad = True
2022-07-28 11:22:37,996 INFO     Parameter score_transform_ent_fun.bias: torch.Size([2]), require_grad = True
2022-07-28 11:22:37,996 INFO     Parameter Number: 4002
2022-07-28 11:22:38,037 INFO     Loading checkpoint models/fb15k-237-betae...
2022-07-28 11:22:38,687 INFO     tasks = 2p
2022-07-28 11:22:38,687 INFO     init_step = 99999
2022-07-28 11:22:38,687 INFO     Start Training...
2022-07-28 11:22:38,687 INFO     learning_rate = 0
2022-07-28 11:22:38,687 INFO     batch_size = 2048
2022-07-28 11:22:38,687 INFO     hidden_dim = 1000
2022-07-28 11:22:38,687 INFO     gamma = 12.000000
posix.uname_result(sysname='Linux', nodename='sweetums-609-4.local', release='3.10.0-1160.49.1.el7.x86_64', version='#1 SMP Tue Nov 30 15:51:32 UTC 2021', machine='x86_64')
Traceback (most recent call last):
  File "main.py", line 632, in <module>
    main(parse_args())
  File "main.py", line 537, in main
    log = KGReasoning.train_step(model, optimizer, train_path_iterator, args, step)
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/models.py", line 674, in train_step
    model.forward(positive_sample=positive_sample,
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/cqd/base.py", line 439, in forward
    scores = d2.query_2p(entity_embeddings=self.embeddings[0],
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/cqd/discrete.py", line 93, in query_2p
    atom2_scores_2d, _ = score_candidates(s_emb=x1_k_emb_2d, p_emb=p2_emb,
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/cqd/discrete.py", line 33, in score_candidates
    atom_scores_2d = scoring_function(s_emb, p_emb, candidates_emb)
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/cqd/base.py", line 654, in scoring_function
    res = self.minmax_normalize(res)
  File "/home/shinaliu/KG/models/kg-reasoning-alpha/cqd/base.py", line 557, in minmax_normalize
    scores_ = scores_.clamp(min=0.0001)
RuntimeError: CUDA out of memory. Tried to allocate 1.77 GiB (GPU 0; 10.76 GiB total capacity; 8.35 GiB already allocated; 1.37 GiB free; 8.37 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
